napari/layers/base/base.py

```diff
@@ -457,10 +459,17 @@ def _mode_setter_helper(self, mode: Union[Mode, str]) -> Mode:
             New mode for the current layer.
 
         """
-        mode = self._modeclass(mode)
+        mode = self._modeclass(mode_in)
+        # Sub-classes can have their own Mode enum, so need to get members
+        # from the specific mode class set on this layer.
+        PAN_ZOOM = getattr(self._modeclass, "PAN_ZOOM", None)
+        TRANSFORM = getattr(self._modeclass, "TRANSFORM", None)
+        assert PAN_ZOOM is not None
+        assert TRANSFORM is not None
```

Comment:
Why is this needed instead of the simple `self._modecalss.PAN_ZOOM`?

Reply:
We need to make the super class a proper generics on `Mode`.
This is one of the only way to make mypy not complain that both:
1) self._modeclass  does not have attribute XXX
2) later on the XXXX is for sure not None.

Other way is to put an # type: ignore.

It's both a  quirk of mypy and the fact that generics are still recent and quite complex to implement so will be a PR of their own. 

see https://github.com/napari/napari/issues/6005 for followup on generics.

Reply:
I see a +1 from you so i'm going to mark as resolved.